tests/kafkatest/tests/client/consumer_protocol_migration_test.py

```diff
@@ -0,0 +1,301 @@
+
+    RANGE = "org.apache.kafka.clients.consumer.RangeAssignor"
+    COOPERATIVE_STICKEY = "org.apache.kafka.clients.consumer.CooperativeStickyAssignor"
+
+    all_consumer_versions = [LATEST_2_1, LATEST_2_3, LATEST_2_4, LATEST_2_5, \
+                             LATEST_3_2, LATEST_3_4, LATEST_3_5, LATEST_3_6, LATEST_3_7, LATEST_3_8, DEV_BRANCH]
+    consumer_versions_supporting_static_membership = [v for v in all_consumer_versions if v >= LATEST_2_3]
+    consumer_versions_supporting_cooperative_sticky_assignor = [v for v in all_consumer_versions if v >= LATEST_2_4]
+
+    def __init__(self, test_context):
+        super(ConsumerProtocolMigrationTest, self).__init__(test_context, num_consumers=5, num_producers=1,
+                                                            num_zk=0, num_brokers=1, topics={
+                self.TOPIC : { 'partitions': self.NUM_PARTITIONS, 'replication-factor': 1 }
+            })
+
+    def bounce_all_consumers(self, consumer, clean_shutdown=True):
+        for node in consumer.nodes:
+            consumer.stop_node(node, clean_shutdown)
+
+        wait_until(lambda: len(consumer.dead_nodes()) == self.num_consumers, timeout_sec=10,
+                   err_msg="Timed out waiting for the consumers to shutdown")
+
+        # Wait until the group becomes empty. We use the 50-second timeout because the
+        # consumer session timeout is 45 seconds.
+        wait_until(lambda: self.group_id in self.kafka.list_consumer_groups(state="empty"),
+                   timeout_sec=50,
+                   err_msg="Timed out waiting for the group to become empty.")
+
+        for node in consumer.nodes:
+            consumer.start_node(node)
+
+        self.await_all_members(consumer)
+        self.await_consumed_messages(consumer)
+
+    def rolling_bounce_consumers(self, consumer, clean_shutdown=True):
+        for node in consumer.nodes:
+            consumer.stop_node(node, clean_shutdown)
+
+            wait_until(lambda: len(consumer.dead_nodes()) == 1,
+                       timeout_sec=self.session_timeout_sec+5,
+                       err_msg="Timed out waiting for the consumer to shutdown")
+
+            consumer.start_node(node)
+
+            self.await_all_members(consumer)
+            self.await_consumed_messages(consumer)
+
+    def set_group_instance_id(self, consumer):
+        consumer.static_membership = True
+        for ind, node in enumerate(consumer.nodes):
+            node.group_instance_id = "migration-test-member-%d" % ind
+
+    def set_consumer_version(self, consumer, version):
+        for node in consumer.nodes:
+            node.version = version
+
+    def assert_group_type(self, type, timeout_sec=10):
+        wait_until(lambda: self.group_id in self.kafka.list_consumer_groups(type=type), timeout_sec=timeout_sec,
+                   err_msg="Timed out waiting to list expected %s group." % type)
+
+    @cluster(num_nodes=8)
+    @matrix(
+        enable_autocommit=[True],
```

Comment:
nit: Should we remove it from the matrix? It seems that we could hardcode it when we call `setup_consumer`.